rxkeystore/src/androidTest/java/com/nexenio/rxkeystore/provider/BaseCryptoProviderTest.java

```diff
@@ -19,28 +19,43 @@
 
     protected static final String LOREM_IPSUM_LONG = "Cras pharetra pulvinar interdum. Integer bibendum neque dolor, quis sodales dui bibendum non. Curabitur eleifend massa eros, sollicitudin porta lacus convallis eget. Mauris molestie vulputate mi. Pellentesque sit amet tortor a justo tincidunt viverra faucibus a erat. Praesent nec ante eget leo placerat congue eget id elit. Vivamus maximus consectetur malesuada. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Sed at bibendum lorem, ut bibendum nunc. Quisque ut justo non augue egestas rhoncus quis vitae neque. Integer feugiat diam sapien, eu tincidunt ipsum rutrum vitae. Cras pharetra pulvinar interdum. Integer bibendum neque dolor, quis sodales dui bibendum non. Curabitur eleifend massa eros, sollicitudin porta lacus convallis eget. Mauris molestie vulputate mi. Pellentesque sit amet tortor a justo tincidunt viverra faucibus a erat. Praesent nec ante eget leo placerat congue eget id elit. Vivamus maximus consectetur malesuada. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Sed at bibendum lorem, ut bibendum nunc. Quisque ut justo non augue egestas rhoncus quis vitae neque. Integer feugiat diam sapien, eu tincidunt ipsum rutrum vitae. Cras pharetra pulvinar interdum. Integer bibendum neque dolor, quis sodales dui bibendum non. Curabitur eleifend massa eros, sollicitudin porta lacus convallis eget. Mauris molestie vulputate mi. Pellentesque sit amet tortor a justo tincidunt viverra faucibus a erat. Praesent nec ante eget leo placerat congue eget id elit. Vivamus maximus consectetur malesuada. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Sed at bibendum lorem, ut bibendum nunc. Quisque ut justo non augue egestas rhoncus quis vitae neque. Integer feugiat diam sapien, eu tincidunt ipsum rutrum.";
 
+    protected static Provider originalProvider;
+    protected static int originalPosition;
+
     protected Context context;
     protected RxKeyStore keyStore;
     protected RxCryptoProvider cryptoProvider;
 
     @CallSuper
     protected void setUpBeforeEachTest() {
-        setupSecurityProviders();
-
         context = InstrumentationRegistry.getInstrumentation().getTargetContext();
         keyStore = createKeyStore();
         cryptoProvider = createCryptoProvider(keyStore);
     }
 
-    protected void setupSecurityProviders() {
-        final Provider provider = Security.getProvider(BouncyCastleProvider.PROVIDER_NAME);
-        if (!(provider instanceof BouncyCastleProvider)) {
+    public static void setupSecurityProviders() {
```

Comment:
Is this test logic or can we move it into the library? It seems to be required in multiple places.

Reply:
Should be possible to move it into the library but I'm not sure if it would work as intended. The idea is to use a `BouncyCastleProvider` of your choice. By moving it into the library, the projects using the library would use the `BouncyCastleProvider` version we include in this library. Therefore all projects should probably copy this code snippet into their own project if I'm not mistaken.